File: array_max.rb
# Write a function that returns the greatest value from an array of numbers.

# Input: [5, 17, -4, 20, 12]
# Output: 20

def array_max(array)
  max = array[0]
  array.each do |number|
    max = number if number > max
  end
  return max
end

p array_max([5, 17, -4, 20, 12])

File: c++/array_max.cpp
// Write a function that returns the greatest value from an array of numbers.

// Input: [5, 17, -4, 20, 12]
// Output: 20

#include <iostream>
using namespace std;

int main()
{
  int input[5] = {5, 17, -4, 20, 12};
  int length = sizeof(input) / sizeof(input[0]);
  int maximum;
  for (int i = 0; i < length; i++)
  {
    if (input[i] > maximum)
    {
      maximum = input[i];
    }
  }
  cout << maximum << endl;
  return 0;
}